### PR TITLE
Build script for summit

### LINKED ISSUE #1478: Verify and update Summit QE6.4 recipe and documentation
After #1457 is merged

### LINKED ISSUE #1712: Build script needed for Summit
Add a build script for Summit to the config directory.  This will ensure users have a less error prone route to a performant build.

### PR DESCRIPTION
**Pull request by PDoakORNL**
WIP since I want to verify whether we want mixed precision for ENABLE_CUDA builds
@ye-luo ?

Closes #1712 

Closes #1478

### CODE REVIEW

File: config/build_olcf_summit.sh [hunk 1]
```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+module purge
+echo "Purging current module set"
+
+echo "Loading QMCPACK dependency modules for summit"
+module load gcc
+module load spectrum-mpi/10.3.0.1-20190611
```
Comment (ye-luo) [inline config/build_olcf_summit.sh]: I feel better not to put a specific version.

**Comment by ye-luo** [comment]
Can we set flags using pattern matching?
tests/test_automation/nightly_olcf_summit.sh

We do need complex/real + FP/MP for ENABLE_CUDA.

Please delete titan build script as well.

**Comment by ye-luo** [comment]
What does mk1 mean?

**Comment by PDoakORNL** [comment]
mark 1 redundant with [WIP]

**Comment by ye-luo** [comment]
Offload is not enabled currently. Enable_CUDA is only useful for AFQMC at the moment. We should call it GPU_AFQMC.
Do we need CPU only builds on Summit?

**Comment by prckent** [comment]
Yes, we need CPU only builds on Summit.

**Comment by prckent** [comment]
As discussed on the ECP call just now, for the release we will accept the minimal working script + same for QE (CPU only) and updated build documentation and refine later.

**Comment by PDoakORNL** [comment]
Added CPU only builds, next I will update documentation and add QE.  Could not find a compiler combination that would build both AFQMC(CPU only) and legacy cuda.

**Comment by prckent** [comment]
(Does need to be addressed now, but we should understand these compiler restrictions for AFQMC and legacy CUDA on Summit. Per oxygen there is not a restriction here...)

**Comment by ye-luo** [comment]
Since the new QE recipe is as simple as what can be used on a linux box and all the modifications to the make.inc for QE6.3 ars no more needed. I prefer removing the QE recipe in the manual and the script. A performant QE need other libraries like ELPA as well.

**Comment by PDoakORNL** [comment]
I found that build problems result on summit unless all the compilers are explicitly set before configuring I believe I previously found FC, F90, F77 needed to be set and for pgi CC needed to be as well.

This is admittedly a big improvement.
```
CXX=g++ CC=gcc FC=gfortran MPIF90=mpif90 F90=gfortran F77=gfortran ./configure --with-hdf5=$OLCF_HDF5_ROOT
```

**Comment by prckent** [comment]
Getting the HDF5 configure right is proven to be not obvious on all platforms. We will have a Summit script even if it is "simple" to some of us.

**Comment by ye-luo** [comment]
@prckent The QE configure script has been improved a lot and for all the machines I use, there is no need of modifying make.inc by hand. The `configure -h` is clear about what is needed.

I think we can mention `--with-hdf5=$OLCF_HDF5_ROOT` and the used compiler in the manual.
If we really would like to provide a QE build script, we must put all the libraries like scalapack, essl.

**Comment by prckent** [comment]
@ye-luo We can improve things later. This is a start.

File: external_codes/quantum_espresso/build_qe_olcf_summit.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+module purge
+echo "Purging current module set"
+. ../../config/load_summit_modules.sh
+
+./download_and_patch_qe6.4.1.sh
+cd qe-6.4.1
+
+CXX=g++ CC=gcc FC=gfortran MPIF90=mpif90 F90=gfortran F77=gfortran ./configure --with-hdf5=$OLCF_HDF5_ROOT
+
+make all
```
[the remark this answers is not in the document]
Comment (PDoakORNL) [reply external_codes/quantum_espresso/build_qe_olcf_summit.sh]: Testing this now.

File: config/build_olcf_summit.sh [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+do
+    echo "building: $build with ${builds[$build]}"
+    rm bin/qmcpack_${build}
+    mkdir build_summit_${build}
+    cd build_summit_${build}
+    cmake -DCMAKE_C_COMPILER="mpicc" \
+      -DCMAKE_CXX_COMPILER="mpicxx" \
+      -DBUILD_LMYENGINE_INTERFACE=0 \
+      ${builds[$build]} \
+      -DCUDA_ARCH="sm_70" \
```
[the remark this answers is not in the document]
Comment (PDoakORNL) [reply config/build_olcf_summit.sh]: Fixed and testing.

**Comment by ye-luo** [comment]
Test this please

File: external_codes/quantum_espresso/build_qe_olcf_summit.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (PDoakORNL) [reply external_codes/quantum_espresso/build_qe_olcf_summit.sh]: This works.

**Pull request merged**